packages/omg-js-util/src/transaction.js

```diff
@@ -271,6 +284,32 @@ const transaction = {
   */
   getToSignHash: function (typedData) {
     return getToSignHash(typedData)
+  },
+
+  /**
+  * Transaction metadata is of type `bytes32`. To pass a string shorter than 32 bytes it must be hex encoded and padded.
+  * This method is one way of doing that.
+  *
+  *@param {string} str the string to be encoded and passed as metadata
+  *@returns the encoded metadata
+  *
+  */
+  encodeMetadata: function (str) {
+    const encodedMetadata = Buffer.from(str).toString('hex').padStart(64, '0')
+    return `0x${encodedMetadata}`
+  },
+
+  /**
+  * Transaction metadata is of type `bytes32`. To pass a string shorter than 32 bytes it must be hex encoded and padded.
+  * This method decodes a string that was encoded with encodeMetadata().
+  *
+  *@param {string} str the metadata string to be decoded
+  *@returns the decoded metadata
+  *
+  */
+  decodeMetadata: function (str) {
+    const unpad = str.replace('0x', '').replace(/^0*/, '')
```

Comment:
Very cool, just to be sure, the `/^0*/` would only remove the left pad leading zeroes and not the zeroes within the hex string itself ?

Reply:
Yep